nltk/parse/corenlp.py

```diff
@@ -0,0 +1,649 @@
+            #       accumulate when lots of text is being parsed.
+            self.popen = java(
+                cmd,
+                classpath=self._classpath,
+                blocking=False,
+                stdout='pipe',
+                stderr='pipe',
+            )
+        finally:
+            # Return java configurations to their default values.
+            config_java(options=default_options, verbose=self.verbose)
+
+        # Check that the server is istill running.
+        # TODO: is there a better way of checking whether a server is ready to
+        #       accept connections?
+        time.sleep(5)
+        returncode = self.popen.poll()
+        if returncode is not None:
+            _, stderrdata = self.popen.communicate()
+            raise CoreNLPServerError(
+                returncode,
+                'Could not start the server. '
+                'The error was: {}'.format(stderrdata.decode('ascii'))
+            )
+
+    def stop(self):
+        self.popen.terminate()
+        self.popen.wait()
+
+    def __enter__(self):
+        self.start()
+
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.stop()
+        return False
+
+
+class GenericCoreNLPParser(ParserI, TokenizerI):
+    """Interface to the CoreNLP Parser."""
+
+    def __init__(self, url='http://localhost:9000', encoding='utf8'):
+
+        self.url = url
+        self.encoding = encoding
+
+        self.session = requests.Session()
+
+    def parse_sents(self, sentences, *args, **kwargs):
+        """Parse multiple sentences.
+
+        Takes multiple sentences as a list where each sentence is a list of
+        words. Each sentence will be automatically tagged with this
+        CoreNLPParser instance's tagger.
+
+        If a whitespace exists inside a token, then the token will be treated as
+        several tokens.
+
+        :param sentences: Input sentences to parse
+        :type sentences: list(list(str))
+        :rtype: iter(iter(Tree))
+        """
+
+        sentences = (' '.join(words) for words in sentences)
+        return self.raw_parse_sents(sentences, *args, **kwargs)
+
+    def raw_parse(self, sentence, properties=None, *args, **kwargs):
+        """Parse a sentence.
+
+        Takes a sentence as a string; before parsing, it will be automatically
+        tokenized and tagged by the CoreNLP Parser.
+
+        :param sentence: Input sentence to parse
+        :type sentence: str
+        :rtype: iter(Tree)
+        """
+        default_properties = {
+            'tokenize.whitespace': 'false',
+        }
+        default_properties.update(properties or {})
+
+        return next(
+            self.raw_parse_sents(
+                [sentence],
+                properties=default_properties,
+                *args,
+                **kwargs
+            )
+        )
+
+    def api_call(self, data, properties=None):
+        default_properties = {
+            'outputFormat': 'json',
```

Comment:
You can find the Stanford CoreNLP protobuf file [here](https://github.com/stanfordnlp/stanza/blob/master/stanza/nlp/CoreNLP_pb2.py), and check out an implementation that uses the protobuf [here](https://github.com/stanfordnlp/stanza/blob/master/stanza/nlp/corenlp.py).